Zend/zend_compile.c

```diff
@@ -3025,7 +3068,15 @@ void zend_compile_assign(znode *result, zend_ast *ast) /* {{{ */
 				/* list($a, $b) = $a should evaluate the right $a first */
 				zend_compile_simple_var_no_cv(&expr_node, expr_ast, BP_VAR_R, 0);
 			} else {
-				zend_compile_expr(&expr_node, expr_ast);
+				if (zend_is_variable(expr_ast)) {
+					if (zend_compile_list_assign_requires_w(var_ast)) {
+						zend_compile_var(&expr_node, expr_ast, BP_VAR_W);
+					} else {
+						zend_compile_expr(&expr_node, expr_ast);
+					}
+				} else {
+					zend_error_noreturn(E_COMPILE_ERROR, "Cannot assign reference to non referencable value");
```

Comment:
Looks like the branches are inverted here. Now this error will also be thrown for normal, non-reference list assignments.

Reply:
I neglected to run against more than just the list tests.  Yes, it should be checking not if variable, but, if needs a write.